src/multicall/starknet.ts

```diff
@@ -17,67 +17,78 @@ function parseStarknetResult(rawResult: string[], functionAbi: any): any {
     return rawResult;
   }
 
-  const output = functionAbi.outputs[0];
-  const rawValue = rawResult[0];
+  // Parse each output according to its type
+  const outputs = functionAbi.outputs;
+  const results: any[] = [];
+  let rawIndex = 0; // Track position in rawResult array
 
   try {
-    switch (output.type) {
-      case 'core::felt252':
-        // Try to decode as shortString (for name, symbol)
-        try {
-          return shortString.decodeShortString(rawValue);
-        } catch {
-          // If shortString decode fails, return as hex
-          return rawValue;
-        }
-
-      // Unsigned integers
-      case 'core::integer::u8':
-      case 'core::integer::u16':
-      case 'core::integer::u32':
-      case 'core::integer::u64':
-        return parseInt(rawValue, 16);
-
-      case 'core::integer::u128':
-      case 'core::integer::usize':
-        return BigInt(rawValue).toString();
-
-      case 'core::integer::u256':
-        return uint256.uint256ToBN({
-          low: rawValue,
-          high: rawResult[1] || '0x0'
-        });
-
-      // Signed integers
-      case 'core::integer::i8':
-      case 'core::integer::i16':
-      case 'core::integer::i32':
-      case 'core::integer::i64':
-        return parseInt(rawValue, 16);
-
-      case 'core::integer::i128':
-        return BigInt(rawValue).toString();
-
-      // Boolean type
-      case 'core::bool':
-        return rawValue === '0x1' || rawValue === '0x01';
-
-      // Address types
-      case 'core::starknet::contract_address::ContractAddress':
-      case 'core::starknet::class_hash::ClassHash':
-      case 'core::starknet::storage_access::StorageAddress':
-        return rawValue;
-
-      // Byte array
-      case 'core::bytes_31::bytes31':
-        return rawValue;
-
-      default:
-        // Return raw value for unknown types
-        return rawValue;
+    for (let outputIndex = 0; outputIndex < outputs.length; outputIndex++) {
```

Comment:
Are we fine with having this outside the loop ?

This will return the rawResult if any of the parsing fail (will never return partially parsed results)